### PR TITLE
Category page

### PR DESCRIPTION
**Pull request by harshitpaliwal95**
## FEAT:
- add category to database
- categories on home page

### CODE REVIEW

File: src/modal/modal.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+  email: string;
+  password: string;
+  name?: string;
+  lastName?: string;
+}
+export interface CategoryType {
+  _id?: string;
+  categoryName: string;
+  img: string;
+}
```
Comment (darshandamre) [inline src/modal/modal.ts]: why are these types in `modal/modal.ts`, don't make sense to me

**Review by darshandamre — COMMENTED** [review]
good work!

File: src/modal/modal.ts (hunk 1 above)
Comment (harshitpaliwal95) [reply src/modal/modal.ts]: I create a separate file for the types that I'll be using.
and the name should be types.ts

File: src/components/CategoryCard/categoryCard.css [hunk 2]
```diff
@@ -0,0 +1,17 @@
+.card-top-img {
+  object-fit: initial;
+}
+.card-badge {
+  background-color: #100d2a;
```
Comment (athar05) [inline src/components/CategoryCard/categoryCard.css:5]: you can consider using variables.

File: src/pages/homePage/home.tsx [hunk 3]
```diff
@@ -1,12 +1,33 @@
+import axios from "axios";
+import { useEffect, useState } from "react";
 import { CategoryCard } from "../../components";
+import { CategoryType } from "../../types";
 
 export const Home = () => {
+  const [data, setdata] = useState<CategoryType[]>();
```
Comment (athar05) [inline src/pages/homePage/home.tsx:7]: It should have been "setData" as per the convention

**Review by athar05 — COMMENTED** [review]
Overall the code looks clean. Added a few minor sugessions.

**Comment by athar05** [comment]
The UI looks decent. Would love to see how the app comes along.

File: src/types.ts [hunk 4]
```diff
@@ -0,0 +1,11 @@
+export interface AuthBody {
+  email: string;
+  password: string;
+  name?: string;
+  lastName?: string;
+}
+export interface CategoryType {
+  _id?: string;
```
Comment (Pratham82) [inline src/types.ts:8]: If _id is required then don't add optional property in the category type. Apart from this all looks good. Good folder structure.

**Review by Pratham82 — APPROVED** [review]
Good to merge

File: src/types.ts (hunk 4 above)
Comment (harshitpaliwal95) [reply src/types.ts:8]: i am using categoryType in other objs as well where id is not require should i make separate type for that or i can use like this

**Pull request merged**